src/main/java/net/p3pp3rf1y/sophisticatedbackpacks/compat/recipeviewers/emi/BackpackEmiPlugin.java

```diff
@@ -0,0 +1,109 @@
+package net.p3pp3rf1y.sophisticatedbackpacks.compat.recipeviewers.emi;
+
+import dev.emi.emi.api.EmiEntrypoint;
+import dev.emi.emi.api.EmiPlugin;
+import dev.emi.emi.api.EmiRegistry;
+import dev.emi.emi.api.recipe.EmiRecipeCategory;
+import dev.emi.emi.api.recipe.VanillaEmiRecipeCategories;
+import dev.emi.emi.api.stack.EmiStack;
+import dev.emi.emi.api.widget.Bounds;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.level.block.Block;
+import net.p3pp3rf1y.sophisticatedbackpacks.client.gui.BackpackScreen;
+import net.p3pp3rf1y.sophisticatedbackpacks.client.gui.BackpackSettingsScreen;
+import net.p3pp3rf1y.sophisticatedbackpacks.compat.recipeviewers.common.DyeRecipesMaker;
+import net.p3pp3rf1y.sophisticatedbackpacks.init.ModItems;
+import net.p3pp3rf1y.sophisticatedcore.client.gui.SettingsScreen;
+import net.p3pp3rf1y.sophisticatedcore.compat.recipeviewers.emi.EmiClientRecipeHelper;
+import net.p3pp3rf1y.sophisticatedcore.compat.recipeviewers.emi.EmiGridMenuInfo;
+import net.p3pp3rf1y.sophisticatedcore.compat.recipeviewers.emi.EmiSettingsGhostDragDropHandler;
+import net.p3pp3rf1y.sophisticatedcore.compat.recipeviewers.emi.EmiStorageGhostDragDropHandler;
+import net.p3pp3rf1y.sophisticatedcore.compat.recipeviewers.emi.comparison.EmiSubtypeInterpreter;
+
+import java.util.function.Consumer;
+
+import static net.p3pp3rf1y.sophisticatedbackpacks.compat.recipeviewers.common.subtypes.SubtypeInterpreters.getSubtypeInterpreters;
+
+@EmiEntrypoint
+public class BackpackEmiPlugin implements EmiPlugin {
+	private static Consumer<WorkstationRegistration> additionalWorkstations = registrar -> {};
+	public static void addAdditionalWorkstations(Consumer<WorkstationRegistration> additionalWorkstations) {
+		BackpackEmiPlugin.additionalWorkstations = BackpackEmiPlugin.additionalWorkstations.andThen(additionalWorkstations);
+	}
+
+	public static class WorkstationRegistration {
+		private final EmiRegistry registry;
+
+		private WorkstationRegistration(EmiRegistry registry) {
+			this.registry = registry;
+		}
+
+		public void addWorkstation(ResourceLocation id, Block icon, Item workstation) {
+			addWorkstation(new EmiRecipeCategory(id, EmiStack.of(icon)), workstation);
+		}
+
+		public void addWorkstation(EmiRecipeCategory category, Item workstation) {
+			addWorkstation(category, EmiStack.of(workstation));
+		}
+
+		public void addWorkstation(EmiRecipeCategory category, EmiStack workstation) {
+			registry.addWorkstation(category, workstation);
+		}
+	}
+
+    @Override
```

Comment:
there seems to be a bit of formatting issue here (4 spaces instead of tab), not a huge deal and something I can fix later if needed